### ISSUE TITLE
locksmithd: panic on shutdown

### ISSUE THREAD

**Issue by lucab** [coreos/bugs#1987]
As seen from the shutdown logs of an AWS instance running CL alpha-1432.0.0 (locksmith 0.4.2):
```
May 31 14:06:11 ip-172-31-0-51 systemd[1]: Stopping D-Bus System Message Bus...
May 31 14:06:11 ip-172-31-0-51 locksmithd[668]: panic: runtime error: invalid memory address or nil pointer dereference
May 31 14:06:11 ip-172-31-0-51 locksmithd[668]: [signal SIGSEGV: segmentation violation code=0x1 addr=0x30 pc=0x5cb376]
May 31 14:06:11 ip-172-31-0-51 locksmithd[668]: goroutine 16 [running]:
May 31 14:06:11 ip-172-31-0-51 locksmithd[668]: panic(0x91bb00, 0xc42000c060)
May 31 14:06:11 ip-172-31-0-51 locksmithd[668]:         /usr/lib/go1.7/src/runtime/panic.go:500 +0x1a1
May 31 14:06:11 ip-172-31-0-51 locksmithd[668]: github.com/coreos/locksmith/updateengine.(*Client).RebootNeededSignal(0xc4200bfcc0, 0xc42000ea20, 0xc42000e4e0)
May 31 14:06:11 ip-172-31-0-51 locksmithd[668]:         /build/amd64-usr/var/tmp/portage/app-admin/locksmith-0.4.2/work/gopath/src/github.com/coreos/locksmith/updateengine/client.go:83 +0x116
May 31 14:06:11 ip-172-31-0-51 locksmithd[668]: created by main.runDaemon
May 31 14:06:11 ip-172-31-0-51 locksmithd[668]:         /build/amd64-usr/var/tmp/portage/app-admin/locksmith-0.4.2/work/gopath/src/github.com/coreos/locksmith/locksmithctl/daemon.go:388 +0x67d
```

**Comment by dgonyeo** [comment]
I just stumbled across this myself and spent a little time looking at it.

I think one solution might be to add:

```
if signal == nil {
    continue
}
```

here: https://github.com/coreos/locksmith/blob/f351194e191485981a71c638127be025f9b44dc8/updateengine/client.go#L92

but I'm not familiar enough with the codebase to be sure that it wouldn't subtly cause issues. I don't know if receiving nil messages from dbus is indicative of some other problem.

**Comment by rfairley** [comment]
Reproduced the panic message using a QEMU image in the SDK (chroot version 1814.0.0), using the scripts described in https://gist.github.com/rfairley/fe4a25c6abc4df61acc223e3e574a50c.

After adding the change suggested by @dgonyeo and running the same scripts, the panic message no longer appears in the `journal.txt` logs. Running all available kola tests for QEMU (with `kola run -p qemu`) inside the SDK with the change added also showed no panic - so seems like there isn't an issue on the QEMU side.

As I understand there is no need to handle the `nil` signals, so it is fine to simply `continue`. Typically the amount of `nil` signals sent is none, or 20 times (based on using a `println` statement before the continue), so there isn't a great number of `nil` signals generated that cause spinning in the `for` loop.

**Comment by lucab** [comment]
Fixed in locksmith 0.6.2: https://github.com/coreos/locksmith/releases/tag/v0.6.2

CL bump is targeted at next alpha: https://github.com/coreos/coreos-overlay/pull/3314
